### ISSUE TITLE
Tests fail with new Mojolicious (>= 8.03)

### ISSUE THREAD

**Issue by eserte** [dotandimet/Mojo-Feed#16]
The t/24-plugin.t test fails:
```
    #   Failed test 'undef isa 'HASH''
    #   at t/24-plugin.t line 46.
    #     undef isn't defined
    
    #   Failed test at t/24-plugin.t line 47.
    #          got: undef
    #     expected: 'First Weblog'
    
    #   Failed test 'undef isa 'HASH''
    #   at t/24-plugin.t line 52.
    #     undef isn't defined
    
    #   Failed test at t/24-plugin.t line 53.
    #          got: undef
    #     expected: 'First Weblog'
    
    #   Failed test 'undef isa 'HASH''
    #   at t/24-plugin.t line 57.
    #     undef isn't defined
    
    #   Failed test 'string ref from body'
    #   at t/24-plugin.t line 58.
    #          got: undef
    #     expected: 'First Weblog'
Mojo::Transaction::success is DEPRECATED in favor of Mojo::Transaction::result and Mojo::Transaction::error at /home/cpansand/.cpan/build/2018110316/Mojo-Feed-0.16-mhgEZe/blib/lib/Mojo/Feed/Reader.pm line 73.
Error getting feed from url http://127.0.0.1:53713/atom.xml: Not Found at /home/cpansand/.cpan/build/2018110316/Mojo-Feed-0.16-mhgEZe/blib/lib/Mojolicious/Plugin/FeedReader.pm line 62.
    # Child (parse) exited without calling finalize()

#   Failed test 'parse'
#   at /usr/perl5.20.1Dp/lib/site_perl/5.20.1/Test/Builder.pm line 279.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 1.
t/24-plugin.t ...... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/1 subtests 
```

Statistical analysis suggests that this happens with new Mojolicious version:
```
****************************************************************
Regression 'mod:Mojolicious'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	9097471055264232.00
[1='eq_7.57']  	      0.0000	      0.0000	   3.57
[2='eq_7.58']  	     -0.0000	      0.0000	  -1.24
[3='eq_7.62']  	      0.0000	      0.0000	   2.06
[4='eq_7.64']  	      0.0000	      0.0000	   0.82
[5='eq_7.65']  	      0.0000	      0.0000	   1.65
[6='eq_7.68']  	      0.0000	      0.0000	   4.29
[7='eq_7.70']  	      0.0000	      0.0000	   4.96
[8='eq_7.71']  	     -0.0000	      0.0000	  -0.49
[9='eq_7.72']  	      0.0000	      0.0000	   3.91
[10='eq_7.75'] 	      0.0000	      0.0000	   4.37
[11='eq_7.81'] 	      0.0000	      0.0000	   1.75
[12='eq_7.83'] 	      0.0000	      0.0000	   0.00
[13='eq_7.84'] 	      0.0000	      0.0000	   4.37
[14='eq_7.85'] 	     -0.0000	      0.0000	  -1.65
[15='eq_7.88'] 	      0.0000	      0.0000	   0.00
[16='eq_8.0']  	     -0.0000	      0.0000	  -2.86
[17='eq_8.03'] 	     -1.0000	      0.0000	-7878641044052450.00
[18='eq_8.04'] 	     -1.0000	      0.0000	-8674098307939065.00
[19='eq_8.05'] 	     -1.0000	      0.0000	-8973690171151438.00

R^2= 1.000, N= 139, K= 20
****************************************************************
```

Also I see a deprecation warning:
```
Mojo::Transaction::success is DEPRECATED in favor of Mojo::Transaction::result and Mojo::Transaction::error at /home/cpansand/.cpan/build/2018110316/Mojo-Feed-0.16-mhgEZe/blib/lib/Mojo/Feed/Reader.pm line 73.
```

**Comment by dotandimet** [comment]
Thanks! fixed in commit 37b92471

**Comment by dotandimet** [comment]
And the fix was released in version 0.17
